Parsers/ASimNetworkSession/ProductParsers/NetworkSessionMicrosoftWindowsFirawall.yaml

```diff
@@ -13,104 +13,209 @@ References:
+                      EventSeverity=tostring(EventData.Severity),
+                      LayerCode = tostring(EventData.LayerName),
+                      DirectionCode = tostring(EventData.Direction),
+                      NetworkRuleNumber = tostring(EventData.FilterRTID),
+                      Protocol = toint(EventData.Protocol);
+              let OutboundEvents = 
+                  RawFireWallEvents
+                  | lookup Directions on DirectionCode
+                  | where isOutBound
+                  | extend 
+                          SrcAppName = tostring(EventData.Application),
+                          SrcIpAddr = tostring(EventData.SourceAddress),
+                          SrcHostId = tostring(EventData.RemoteMachineID),
+                          SrcPortNumber = toint(EventData.SourcePort),
+                          SrcProcessId = tostring(EventData.ProcessId),
+                          DstPortNumber = toint(EventData.DestPort),
+                          DstIpAddr = tostring(EventData.DestAddress),
+                          DstUserId = tostring(EventData.RemoteUseID),
+                          DstUserIdType = "SID";
+              let InboundEvents = 
+                  RawFireWallEvents
+                  | lookup Directions on DirectionCode
+                  | where not(isOutBound)
+                  | extend 
+                          DstAppName = tostring(EventData.Application),
+                          DstIpAddr	 = tostring(EventData.SourceAddress),
+                          DstHostId = tostring(EventData.RemoteMachineID),
+                          DstPortNumber = toint(EventData.SourcePort),
+                          DstProcessId = tostring(EventData.ProcessID),
+                          SrcPortNumber = toint(EventData.DestPort),
+                          SrcIpAddr = tostring(EventData.DestAddress),
+                          SrcUserId = tostring(EventData.RemoteUseID),
+                          SrcUserIdType = "SID";
+              union isfuzzy=true InboundEvents, OutboundEvents
+              };
+        // Main query -> outputs both schemas as one normalized table
+        union isfuzzy=true WindowsFirewall_SecurityEvent, WindowsFirewall_WindowsEvent
+            | extend 
+                    DvcAction = iff(EventID in (5154, 5156, 5158), "Allow", "Deny"),
+                    DvcHostname = Computer,
+                    DvcOs = 'Windows',
+                    EventType = "NetworkSession",
+                    EventSchema = "NetworkSession",
+                    EventSchemaVersion="0.2.0",
+                    EventCount=toint(1),
+                    EventVendor = "Microsoft",
+                    EventProduct = "WindowsFirewall",
+                    EventResult = "Success",
+                    EventStartTime = TimeGenerated,
+                    EventEndTime = TimeGenerated,
+                    EventOriginalUid = EventOriginId,
+                    EventSeverity = "Informational"
```

Comment:
I think blocked events should not be informational. What do you think?

Reply:
Again, a philosophic question: on its own, the event of blocking a packet or a connection is not a threat: how can we score its severity?

Reply:
A good question: Thoughts: should i add NA? Make EventSeverity not applicable? or just have those as "Low". We can chat tomorrow morning on that.

Reply:
I think it should be something like NA since i believe that customers will create alerts based on this field